monailabel/tasks/train/basic_train.py

```diff
@@ -157,7 +157,7 @@ def __init__(
         self._config = {
             "name": "train_01",
             "pretrained": True,
-            "device": "cuda",
+            "device": device_displayname_list()[0],
```

Comment:
here you set the displayname to the GPU model names, but didn't change back to cuda when send the request to server, pytorch still need the device ordinal to be "cuda". You may need similar changes as you did in the basic_infer.py